cedar-lean/Cedar/Thm/Validation.lean

```diff
@@ -0,0 +1,46 @@
+import Cedar.Spec
+import Cedar.Data
+import Cedar.Validation
+import Cedar.Thm.Validation.Validator
+
+namespace Cedar.Thm
+
+open Cedar.Data
+open Cedar.Spec
+open Cedar.Validation
+
+/--
+If validation succeeds, then for any request consistent with the schema, either
+(1) evaluation of every policy produces a boolean or (2) returns an error TODO
+-/
+theorem validation_is_sound (policies : Policies) (schema : Schema) (request : Request) (entities : Entities) :
+validate policies schema = .ok () →
+(∀ env ∈ schema.toEnvironments,
+RequestAndEntitiesMatchEnvironment env request entities) →
```

Comment:
Might be nice to pull this out into a new predicate `RequestAndEntitiesMatchSchema`